middleware/locale/locale_test.go

```diff
@@ -0,0 +1,53 @@
+package locale_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+func TestLocale(t *testing.T) {
+	tests := []struct {
+		locales              []string
+		methods              []method.Method
+		acceptLanguageHeader string
+		expectedLocale       string
+	}{
+		{[]string{"en"}, []method.Method{method.Names["header"]}, "", "en"},
+		{[]string{"en", "en-GB"}, []method.Method{method.Names["header"]}, "en-GB,en", "en-GB"},
+		{[]string{"en", "de"}, []method.Method{method.Names["header"]}, "de,en;q=0.8,en-GB;q=0.6", "de"},
+	}
```

Comment:
```diff
+ {[]string{"en", "de"}, []method.Method{method.Names["header"]}, "en-GB;q=0.6,de", "de"},
```

… and again, don't use ```method``` and/or ```method.Method```. Use a name known from a design-pattern, or something descriptive.